### PR TITLE
Group buttons under 1 tag with type differentiation

### PR DESCRIPTION
**Pull request by wirednkod**
Group buttons under 1 tag (< Button / >) that differentiate by the type attribute. The possible values that the button can take are: "help", "mono",  "monoInvert", "option", "primary",  "primaryInvert",  "secondary",  "submit",  "submitInvert", "tab", "tertiary", "text";

These types replace returns the correct Buttons with the correct props;
E.g the button `<ButtonHelp />` would now be `<Button type="help" />`. 
Rest of props and checks remain the same.
E.g.: 
```jsx
<ButtonPrimary text="Button" iconLeft={faUser} marginRight colorSecondary />
```
would be now:
```jsx
<Button type="primary" text="Button" iconLeft={faUser} marginRight colorSecondary />
```

This way the different button imports can now be reduced from 9 to 1 and the "types" of the buttons are hard typed as can be seen below:

image

### CODE REVIEW

File: packages/core-ui/lib/index.tsx [hunk 1]
```diff
@@ -48,18 +48,21 @@ export { ModalSeparator } from "./modal/ModalSeparator";
...
+// export { ButtonSecondary } from "./buttons/ButtonSecondary";
+// export { ButtonTertiary } from "./buttons/ButtonTertiary";
+// export { ButtonMono } from "./buttons/ButtonMono";
+// export { ButtonMonoInvert } from "./buttons/ButtonMonoInvert";
+// export { ButtonSubmitInvert } from "./buttons/ButtonSubmitInvert";
+// export { ButtonText } from "./buttons/ButtonText";
+// export { ButtonSubmit } from "./buttons/ButtonSubmit";
+// export { ButtonHelp } from "./buttons/ButtonHelp";
+// export { ButtonTab } from "./buttons/ButtonTab";
+// export { ButtonOption } from "./buttons/ButtonOption";
```
Comment (rossbulat) [inline packages/core-ui/lib/index.tsx:65]: I would still like to use these exports in the staking dashboard, I would like to think the standalone `Button` component as an optional utility. XCM dashboard, absolutely let's use it there, but would like things to remain the same in staking for the timebeing.

File: packages/core-ui/lib/buttons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  ButtonSubmitInvert,
+  ButtonSubmitInvertProps,
+} from "./ButtonSubmitInvert";
+import { ButtonText } from "./ButtonText";
+import { ButtonSubmit, ButtonSubmitProps } from "./ButtonSubmit";
+import { ButtonHelp, ButtonHelpProps } from "./ButtonHelp";
+import { ButtonTab, ButtonTabProps } from "./ButtonTab";
+import { ButtonOption, ButtonOptionProps } from "./ButtonOption";
+
+export type ButtonProps = ComponentBaseWithClassName &
```
Comment (rossbulat) [inline packages/core-ui/lib/buttons/index.tsx:27]: Perhaps just create a giant type:

```
export type ButtonType =
  | "help"
  | "mono"
  | "monoInvert"
  | "option"
  | "primary"
  | "primaryInvert"
  | "secondary"
  | "submit"
  | "submitInvert"
  | "tab"
  | "tertiary"
  | "text";

export type ButtonProps = ComponentBaseWithClassName & {
  // the type of button
  type: ButtonType;
} & (
    | ButtonHelpProps
    | ButtonMonoProps
    | ButtonMonoProps
    | ButtonOptionProps
    | ButtonPrimaryInvertProps
    | ButtonSecondaryProps
    | ButtonSubmitProps
    | ButtonSubmitInvertProps
    | ButtonTabProps
    | ButtonTertiaryProps
    | ButtonMonoProps
    | ButtonPrimaryProps
  );

```

File: packages/core-ui/lib/buttons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/* @license Copyright 2023 @paritytech/polkadot-cloud authors & contributors
```
Comment (rossbulat) [inline packages/core-ui/lib/buttons/index.tsx:1]: I am getting the inkling that this entire file can be automatically built in the rollup process. We then do not need to update this file at all when new buttons are introduced.

I have began exploring custom rollup plugins with the [`rollup-plugin-remove-lines`](https://github.com/paritytech/polkadot-cloud/blob/main/packages/core-ui/plugins/rollup-plugin-remove-lines.js) file, which simply goes through the library and removes the provided lines if found. I have used it in `core-ui` to remove broken CSS imports. It is called at the beginning of the pipeline at the `writeBundle` phase.

I've also [created an issue](https://github.com/paritytech/polkadot-cloud/issues/286) to automatically generate `lib/index.tsx` so we don't have to manually input components.

All for different PRs, but a direction we should definitely head in.

File: packages/core-ui/lib/types.ts [hunk 4]
```diff
@@ -24,8 +24,24 @@ export type ComponentBaseWithClassName = ComponentBase & {
   className?: string;
 };
 
+type ButtonType =
```
Comment (rossbulat) [inline packages/core-ui/lib/types.ts:27]: 👍 nice, ignore my duplication of this above :)

File: packages/core-ui/lib/index.tsx (hunk 1 above)
Comment (wirednkod) [reply packages/core-ui/lib/index.tsx:65]: That was my understanding, thus I added the "triage" title. Will uncomment, and once this is released, we could alter the Dashboard as well following the above pattern

File: packages/core-ui/lib/buttons/index.tsx (hunk 2 above)
Comment (wirednkod) [reply packages/core-ui/lib/buttons/index.tsx:27]: `ButtonType` exists in `packages/core-ui/lib/types.ts`. The `ButtonProps` also in a different format in order to reduce unnecessary repetitions;

File: packages/core-ui/lib/buttons/index.tsx (hunk 3 above)
Comment (wirednkod) [reply packages/core-ui/lib/buttons/index.tsx:1]: My only concern is if there should be internal components that we prefer not to export automatically through this automated process; But lets move the conversation in the newly open PR

**Review by rossbulat — APPROVED** [review]

**Comment by rossbulat** [comment]
Nice addition, and identified areas of automation too :) Feel free to bump the version to trigger a publish when you merge.

**Pull request merged**